app/services/user_updater.rb

```diff
@@ -66,9 +66,6 @@ def update(attributes = {})
       attributes[:title] != user.title &&
       guardian.can_grant_title?(user, attributes[:title])
       user.title = attributes[:title]
-      if user.badges.where(name: user.title).exists?
-        user_profile.badge_granted_title = true
-      end
```

Comment:
existing specs to make sure removing this does not regress:

https://github.com/discourse/discourse/blob/d30e45a45832f29cfd41ea4cc144d0032f43c2f7/spec/services/user_updater_spec.rb#L204-L219